source/assets/js/components/redirect.ts

```diff
@@ -1,410 +1,350 @@
-      '#new-types-argb':
-        '/documentation/js-api/classes/types.Color#constructor',
-      '#color-getr': '/documentation/js-api/classes/types.Color#getR',
-      '#color-getg': '/documentation/js-api/classes/types.Color#getG',
-      '#color-getb': '/documentation/js-api/classes/types.Color#getB',
-      '#color-geta': '/documentation/js-api/classes/types.Color#getA',
-      '#color-setr-red': '/documentation/js-api/classes/types.Color#setR',
-      '#color-setg-green': '/documentation/js-api/classes/types.Color#setG',
-      '#color-setb-blue': '/documentation/js-api/classes/types.Color#setB',
-      '#color-seta-alpha': '/documentation/js-api/classes/types.Color#setA',
-      '#types-boolean': '/documentation/js-api/classes/types.Boolean',
-      '#types-boolean-true': '/documentation/js-api/classes/types.Boolean#TRUE',
-      '#types-boolean-false':
-        '/documentation/js-api/classes/types.Boolean#FALSE',
-      '#types-boolean-getvalue':
-        '/documentation/js-api/classes/types.Boolean#getValue',
-      '#types-list': '/documentation/js-api/classes/types.List',
-      '#new-types-list-length-comma-true':
-        '/documentation/js-api/classes/types.List#constructor',
-      '#list-getvalue-index':
-        '/documentation/js-api/classes/types.List#getValue',
-      '#list-getseparator':
-        '/documentation/js-api/classes/types.List#getSeparator',
-      '#list-getlength': '/documentation/js-api/classes/types.List#getLength',
-      '#list-setvalue-index-value':
-        '/documentation/js-api/classes/types.List#setValue',
-      '#list-setseparator-comma':
-        '/documentation/js-api/classes/types.List#setSeparator',
-      '#types-map': '/documentation/js-api/classes/types.Map',
-      '#new-types-map-length':
-        '/documentation/js-api/classes/types.Map#constructor',
-      '#map-getkey-index': '/documentation/js-api/classes/types.Map#getKey',
-      '#map-getvalue-index': '/documentation/js-api/classes/types.Map#getValue',
-      '#map-getlength': '/documentation/js-api/classes/types.Map#getLength',
-      '#map-setkey-index-key': '/documentation/js-api/classes/types.Map#setKey',
-      '#map-setvalue-index-value':
-        '/documentation/js-api/classes/types.Map#setValue',
-      '#types-null': '/documentation/js-api/classes/types.Null',
-      '#types-null-null': '/documentation/js-api/classes/types.Null#NULL',
-    };
-
-    const redirect: string | undefined = redirects[window.location.hash];
-    if (redirect) {
-      window.location.href = redirect;
-    }
-  }
+const { pathname, hash } = window.location;
+const match = redirects[pathname]?.[hash];
+if (match) {
+  window.location.href = match;
+} else if (pathname === '/documentation/' && hash.match(/-option$/)) {
+  // If the user is looking for a Ruby Sass option, redirect them to the Ruby
+  // Sass page for an explanation that it's deprecated.
+  window.location.href = '/ruby-sass';
```

Comment:
I went with nested records here, seemed easier to understand and maintain. There's a single edge case (that I could find) that uses regex, so I handled it here

Reply:
Looks great!